### PR TITLE
Remove hash alt interface

### PR DESCRIPTION
**Pull request by tom-daubney-arm**
## Description

Remove hash ALT interfaces.

This PR is made up of commits split out from #9226 in order to ease review.

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** Will be provided at end of PR series
- [x] **3.6 backport** not required
- [x] **2.28 backport** not required
- [x] **tests** existing tests suffice

## Notes for the submitter

Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
checklist for PR contributors.

Help make review efficient:
* Multiple simple commits
  - please structure your PR into a series of small commits, each of which does one thing
* Avoid force-push
  - please do not force-push to update your PR - just add new commit(s)
* See our [Guidelines for Contributors](https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/) for more details about the review process.

### CODE REVIEW

File: include/mbedtls/mbedtls_config.h [hunk 1]
```diff
@@ -2603,10 +2602,10 @@
  * Enable the CMAC (Cipher-based Message Authentication Code) mode for block
  * ciphers.
  *
- * \note When #MBEDTLS_CMAC_ALT is active, meaning that the underlying
- *       implementation of the CMAC algorithm is provided by an alternate
- *       implementation, that alternate implementation may opt to not support
- *       AES-192 or 3DES as underlying block ciphers for the CMAC operation.
+ * \note When the underlying implementation of the CMAC algorithm is provided
```
Comment (ronald-cron-arm) [inline include/mbedtls/mbedtls_config.h:2603]: Just remove I think.

File: include/mbedtls/cmac.h [hunk 2]
```diff
@@ -86,7 +80,7 @@ struct mbedtls_cmac_context_t {
  *                      CMAC computation, use mbedtls_cipher_cmac_finish().
  *
  * \note                When the CMAC implementation is supplied by an alternate
- *                      implementation (through #MBEDTLS_CMAC_ALT), some ciphers
+ *                      implementation (through a PSA driver), some ciphers
```
Comment (ronald-cron-arm) [inline include/mbedtls/cmac.h:83]: Just remove the note I think. Ditto below.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

File: include/mbedtls/cmac.h [hunk 3]
```diff
@@ -79,12 +79,6 @@ struct mbedtls_cmac_context_t {
  *                      To start a CMAC computation using the same key as a previous
  *                      CMAC computation, use mbedtls_cipher_cmac_finish().
  *
- * \note                When the CMAC implementation is supplied by an alternate
```
Comment (ronald-cron-arm) [inline include/mbedtls/cmac.h:88]: Two similar notes to remove below as well.

**Review by ronald-cron-arm — CHANGES_REQUESTED** [review]

**Review by ronald-cron-arm — DISMISSED** [review]
LGTM, thanks.

**Review by bensze01 — APPROVED** [review]
LGTM. Please rebase to resolve the conflicts.

**Review by ronald-cron-arm — DISMISSED** [review]
I've checked the rebase, LGTM.

**Comment by tom-daubney-arm** [comment]
@ronald-cron-arm Rebased without needing manual conflict resolution.

**Review by ronald-cron-arm — APPROVED** [review]
LGTM

**Review by bensze01 — APPROVED** [review]
LGTM

**Review by ronald-cron-arm — APPROVED** [review]
Rebase LGTM.

**Review by bensze01 — APPROVED** [review]
LGTM

**Pull request merged**